src/keystore/InMemoryKeystore.ts

```diff
@@ -197,27 +200,50 @@ export default class InMemoryKeystore implements Keystore {
       req.requests,
       async ({ payload, timestampNs }) => {
         const sealed = SealedInvitation.fromBytes(payload)
+        if (sealed.v1) {
+          const headerTime = sealed.v1.header.createdNs
+          if (!headerTime.equals(timestampNs)) {
+            throw new Error('envelope and header timestamp mismatch')
+          }
+
+          const isSender = sealed.v1.header.sender.equals(
+            this.v2Keys.getPublicKeyBundle()
+          )
 
-        const headerTime = sealed.v1.header.createdNs
-        if (!headerTime.equals(timestampNs)) {
-          throw new Error('envelope and header timestamp mismatch')
-        }
-
-        const isSender = sealed.v1.header.sender.equals(
-          this.v2Keys.getPublicKeyBundle()
-        )
-
-        const invitation = await sealed.v1.getInvitation(this.v2Keys)
-        const topicData = {
-          invitation,
-          createdNs: sealed.v1.header.createdNs,
-          peerAddress: isSender
-            ? await sealed.v1.header.recipient.walletSignatureAddress()
-            : await sealed.v1.header.sender.walletSignatureAddress(),
-        }
-        toAdd.push(topicData)
-        return {
-          conversation: topicDataToConversationReference(topicData),
+          const invitation = await sealed.v1.getInvitation(this.v2Keys)
+          const topicData = {
+            invitation,
+            createdNs: sealed.v1.header.createdNs,
+            peerAddress: isSender
+              ? await sealed.v1.header.recipient.walletSignatureAddress()
+              : await sealed.v1.header.sender.walletSignatureAddress(),
+          }
+          toAdd.push(topicData)
+          return {
+            conversation: topicDataToConversationReference(topicData),
+          }
+        } else if (sealed.v2) {
+          const headerTime = sealed.v2.header.createdNs
+          if (!headerTime.equals(timestampNs)) {
+            throw new Error('envelope and header timestamp mismatch')
+          }
+
+          const invitation = await sealed.v2.getInvitation(
+            PrivateKeyBundleV3.fromLegacyBundle(
+              this.v2Keys,
+              AccountLinkedRole.INBOX_KEY
+            )
```

Comment:
This is going to be problematic with InboxKey Rotation, but I'll take that one on. It's going to be a messy refactor given PrivateKeyBundleV1s can no longer be converted to PrivateKeyBundleV3s.